zebra-chain/src/transaction/unmined/zip317.rs

```diff
@@ -198,6 +202,11 @@ pub fn mempool_checks(
     // In zcashd this is `DEFAULT_MIN_RELAY_TX_FEE` and `LEGACY_DEFAULT_FEE`:
     // <https://github.com/zcash/zcash/blob/f512291ff20098291442e83713de89bcddc07546/src/main.h#L71-L72>
     // <https://github.com/zcash/zcash/blob/9e856cfc5b81aa2607a16a23ff5584ea10014de6/src/amount.h#L35-L36>
+    //
+    // ## Note
+    //
+    // If the check above, which checks the number of unpaid actions, doesn't fail, then there is no
+    // way for the legacy check below to fail. This renders the legacy check redundant.
```

Comment:
Optional/Nitpick: We could skip the next check and the last part of `select_mempool_transactions()` when producing block templates if the unpaid actions limit is 0.

```suggestion
    // If the check above for the maximum number of unpaid actions passes with the default 
    // unpaid action limit, then there is no way for the legacy check below to fail. 
    // This renders the legacy check redundant with the default unpaid action limit of 0.
```

Reply:
I'll try to rephrase this again.

Reply:
Done in [b266e40](https://github.com/ZcashFoundation/zebra/pull/8638/commits/b266e4052ad65533d9dc068c049bdf71294a9b06).